### DISCUSSION TITLE
Task<T> GetJsonAsync<T>(string uri) return null with table has a large records

### DISCUSSION THREAD

**Discussion by Loclhh** [oqtane/oqtane.framework#1804] [Q&A] [answered]
I have a table with more than 10 columns and it has more than 37,000 records. In index.razor of the client project I got return null when OnInitializedAsync() with my code: _Customers = await CustomerService.GetCustomersAsync(ModuleState.ModuleId); but it work good when I get fewer records like 1000.

There is any way that I could get all records from a large table?

Thank you,

**Comment by hishamco** [comment]
Can you write a unit test that simulate what's happen, so we can got a fixed if it fails

**Reply by Loclhh** [reply]
I've just download the latest version of oqtane (3.0.0) to try with the new version. If the error happen again, I will do as you suggest. Thank you very much for your reply.

**Comment by sbwalker** [comment]
@Loclhh in a distributed web application you should NEVER send 37,000 records from the server to the client. If you have a high volume of data then you need to break it into smaller groupings based on the needs of your application. For example in this case, it would make sense to require the user to provide some type of filter on the client-side so that only customer records which match the filter are returned by the server API. A filter might be a portion of a customer's name, or a location, or any other attribute which identifies a smaller group of customers.

**Reply by Loclhh** [reply]
Thanks sbwalker, I know that the way I should do. With this situation, I just want to test the oqtane framework as well as its capacity and its possible bugs.

**Reply by sbwalker** [reply]
thank you for the clarification... however just to be clear, an inability to load 37,000 records in a single request is not an Oqtane issue - it is an architectural approach issue in the custom module and perhaps a limitation in the underlying technologies such as EF Core, System.Text.Json, HttpClient, etc...

**Reply by Loclhh** [reply]
Thank sbwalker,
This infomation is very useful for me!

**Comment by hishamco** [comment, marked as answer]
Simply you can use `Skip` & `Take` to make your work done
